app/concepts/base/repository.rb

```diff
@@ -0,0 +1,64 @@
+module Repositories
+    class Base
+        attr_reader :adapter
+
+        def initialize(adapter)
+            @adapter = adapter
+        end
+    
+
+        def exists?(:params)
+          adapter.exists?(:params)
+        end
+
+        def all
+            adapter.all
+        end
+
+        def make_reservation(ticket_params)
```

Comment:
It is a base repository, so other repositories gonna inhertie from this class, and do you think if all models has to know how to make a reservation or checks if ticket is available? You can just put the method for specific model inside the specific repository after calling super().

Btw. I am not sure how do your models work, but I guess you can consider this. ;)